concreteproperties/stress_strain_profile.py

```diff
@@ -457,6 +457,367 @@ def __post_init__(
+                    + 2.254 * (1 + 7.94 * f_l / self.compressive_strength) ** 0.5
+                    - 2 * f_l / self.compressive_strength
+                )
+        else:
+            # calculate unconfined concrete strength
+            f_cc = self.compressive_strength
+
+        # calculate strain associated with max confined/unconfined concrete strength
+        eps_cc = self.eps_co * (1 + 5 * (f_cc / self.compressive_strength - 1))
+
+        # calculate maximum confined/unconfined compressive strain
+        if self.conc_confined:
+            eps_c_max = min(
+                0.004
+                + self.n_steel_strain * rho_st * self.trans_f_y * self.eps_su / f_cc,
+                0.05,
+            )
+        else:
+            eps_c_max = self.eps_c_max_unconfined
+
+        # calculate secant modulus
+        E_sec = f_cc / eps_cc
+
+        if self.conc_confined:
+            self.strains = np.linspace(0, eps_c_max, self.n_points)
+            # add eps_cc point corresponding to max stress point at end
+            self.strains = np.append(self.strains, eps_cc)
+        else:
+            self.strains = np.linspace(0, min(2 * eps_cc, eps_c_max), self.n_points)
+            # add eps_cc point corresponding to max stress point at end
+            self.strains = np.append(self.strains, eps_cc)
+
+        # sort strains numerically
+        self.strains.sort()
+
+        # calculate stresses from strains
+        r = self.elastic_modulus / (self.elastic_modulus - E_sec)
+        x = self.strains / eps_cc
+        self.stresses = f_cc * x * r / (r - 1 + x**r)
+
+        # add spalling branch if specified for unconfined curve
+        if not self.conc_confined and self.conc_spalling:
+            self.strains = np.append(self.strains, self.eps_sp)
+            self.stresses = np.append(self.stresses, 0)
+
+        # calculate max tension strain based on modulus of rupture/concrete tension
+        # strength
+        eps_t = self.tensile_strength / self.elastic_modulus
+
+        if self.conc_tension:
+            # add tension stress/strain limit
+            self.strains = np.insert(self.strains, 0, -eps_t)
+            self.stresses = np.insert(self.stresses, 0, -self.tensile_strength)
+            self.strains = np.insert(self.strains, 0, self.strains[0])
+            self.stresses = np.insert(self.stresses, 0, 0)
+            self.strains = np.insert(self.strains, 0, 2 * self.strains[0])
+            self.stresses = np.insert(self.stresses, 0, 0)
+        else:
+            # add flat horizontal tension stress/strain branch
+            self.strains = np.insert(self.strains, 0, -eps_t)
+            self.stresses = np.insert(self.stresses, 0, 0)
+
+        # initiate ultimate compressive strain as maximum strain
+        self.ultimate_strain = max(self.strains)
+
+        # add small horizontal compressive strain to improve interpolation
+        self.strains = np.append(self.strains, self.strains[-1] + 1e-12)
```

Comment:
As above.

Reply:
as per above

See commit bc11ce88828df3cfcc572478fe8aa87d542ededd for revised code to address `ModifiedMander` review coments, converting numpy array to list as soon as practical so output type is List as intended.